numpy/core/tests/test_longdouble.py

```diff
@@ -30,8 +30,11 @@ def test_scalar_extraction():
 
 
 def test_repr_roundtrip():
-    o = 1 + np.finfo(np.longdouble).eps
-    assert_equal(np.longdouble(repr(o)), o,
+    o = 1 + LD_INFO.eps
+    # We will only see eps in repr if within printing precision.
+    # Printing precision defined as 17 in scalartypes.c.src
+    expected = 1.0 if LD_INFO.precision > 17 else o
+    assert_equal(np.longdouble(repr(o)), expected,
                  "repr was %s" % repr(o))
```

Comment:
I think it's desirable that `repr` roundtrips properly, and I don't think that changing the test to allow the failure here is the right solution. So yes, on PPC64 I think that it should print 31 digits for repr - there's always `str` for the sane-length output.

Obviously it's not desirable use 31 digits on platforms that don't support it, but as a quick fix, I think that 20 should be become 33 (precision +2 used elsewhere).

Adapting `scalartypes.c.src` to choose more intelligently is an obvious next step.